core/api_token_api.php

```diff
@@ -0,0 +1,162 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * API Token API
+ *
+ * @package CoreAPI
+ * @subpackage ApiTokenAPI
+ * @copyright Copyright 2000 - 2002  Kenzaburo Ito - kenito@300baud.org
+ * @copyright Copyright 2002  MantisBT Team - mantisbt-dev@lists.sourceforge.net
+ * @link http://www.mantisbt.org
+ *
+ * @uses crypto_api.php
+ */
+
+require_api( 'crypto_api.php' );
+
+define( 'API_TOKEN_LENGTH', 32 );
+
+/**
+ * Create an API token
+ *
+ * @param string $p_token_name The name (description) identifying what the token is going to be used for.
+ * @param integer $p_user_id The user id.
+ * @return string The plain token.
+ * @access public
+ */
+function api_token_create( $p_token_name, $p_user_id ) {
+	if( is_blank( $p_token_name ) ) {
+		error_parameters( lang_get( 'token_name' ) );
+		trigger_error( ERROR_EMPTY_FIELD, ERROR );
+	}
+
+	$t_token_name = trim( $p_token_name );
+
```

Comment:
Added validation for max length.  Though, we don't typically do that for other APIs.

I'm not sure about the benefit for enforcing uniqueness for token names.  For example, I may create two tokens to use for the same purpose.  Hence, same name makes sense.  What is really going to differentiate them is the creation timestamp and last used timestamp.  So I'm not sure there is value of having the user try to use "token for iPhone app 2" just to make it unique.